In [81]:
pip install bitarray


     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 288.3/288.3 kB 5.8 MB/s eta 0:00:00


In [118]:
from PIL import Image
from bitarray import bitarray

image_path = '102.png'

with Image.open(image_path) as img:
    binary_str = ''.join(format(byte, '08b') for row in img.getdata() for byte in row)

image_bits = bitarray(binary_str)

In [180]:
result = []
result.append('')
for i in range(0, 250000, 8):
    result[-1] += str(binary_str[i:i+8][-2:])
    if len(result[-1]) == 8:
      result.append('')

In [181]:
text = ''
for a in result:
  text += (chr(int(a, 2)))


In [182]:
text = text.replace("'", '')
text = text.replace('\\n', '')
text = text.replace('\n', '')

In [185]:
text[3377:6765]

' KPMG Building Societies Database 2008,)("Kent Reliance, formerly known as Kent Reliance Building Society and also known as KRBS, is a banking services provider based in Kent, England. Until 31 January 2011 KRBS stood for Kent Reliance Building Society, which was created in 1986 and incorporated Chatham Reliance (established 1898), Dover District (established 1861), Herne Bay (established 1888) and Kent and Canterbury (established 1847). On 1 February 2011, Kent Reliance Building Society transferred its business to a new bank, OneSavings Bank plc (which trades as Kent Reliance).KRBS was the fastest growing building society in the UK between 2003 and 2008 and was a member of the Building Societies Association. The dramatic growth of KRBS was driven in part by the establishment in 2002 of a wholly owned subsidiary, Jersey Home Loans Limited, to buy the Jersey mortgage business of Standard Chartered Grindlays Bank Ltd. By September 2008, this subsidiary had mortgages on its books of over

 KPMG Building Societies Database 2008,)("Kent Reliance, formerly known as Kent Reliance Building Society and also known as KRBS, is a banking services provider based in Kent, England. Until 31 January 2011 KRBS stood for Kent Reliance Building Society, which was created in 1986 and incorporated Chatham Reliance (established 1898), Dover District (established 1861), Herne Bay (established 1888) and Kent and Canterbury (established 1847). On 1 February 2011, Kent Reliance Building Society transferred its business to a new bank, OneSavings Bank plc (which trades as Kent Reliance).KRBS was the fastest growing building society in the UK between 2003 and 2008 and was a member of the Building Societies Association. The dramatic growth of KRBS was driven in part by the establishment in 2002 of a wholly owned subsidiary, Jersey Home Loans Limited, to buy the Jersey mortgage business of Standard Chartered Grindlays Bank Ltd. By September 2008, this subsidiary had mortgages on its books of over £700 million, achieved by competing on both price and service in a market dominated by the major high street banks. KRBS also operates a lending business in Guernsey and the Channel Islands operations in aggregate constitute 43% of the Groups mortgage assets.KRBS was also notable as the only member of the UK building society sector to have offshored administrative work to India, via its wholly owned subsidiaries, Easiprocess and EasiOption.  The KRBS Group employed significantly more staff in India than it did in the UK, as a result KRBS had the lowest costs (as a percentage of assets) of any UK building society.",)("OneSavings Bank plc is a British banking company that has been formed by Kent Reliance Building Society and JC Flowers. It is listed on the London Stock Exchange and is a constituent of the FTSE 250 Index.In July 2010 JC Flowers announced its intention to purchase a stake of Kent Reliance, forming OneSavings Plc. The transfer of the business of the society to OneSavings was approved by a majority of members in November 2010. Kent Reliance Building Society converted into a provident society, Kent Reliance Provident Society, which owns a 59.9% stake in OneSavings Bank, withJC Flowers owning the remaining 40.1%. The transfer became effective on 1 February 2011 and OneSavings plc was renamed OneSavings Bank plc.OneSavings Bank plc, also known as OSB, began trading as a bank on 1 February 2011. OSB is a specialist lending and retail savings group authorised by the Prudential Regulation Authority, part of the Bank of England, and regulated by the Financial Conduct Authority and Prudential Regulation Authority. OSB operates through specialist brokers and independent financial advisors in sub-sectors of the lending market. These sub-sectors include Residential Mortgages (comprising first charge, second charge and shared ownership), Buy to let/SME and Personal Loans. The bank is predominantly funded by retail savings originated through the Kent Reliance franchise.",)("KRBS sponsored south London football club Charlton Athletic F.C. under a three-year sponsorship deal from 2009/10-2011/12.  It previously sponsored its more local team Gillingham F.C. for the 2007/08-2009/10 seasons, but this sponsorship ended at the same time that Gillingham were relegated from League One to League Two.  The team shirts bear KRBSs website address, krbs.com.",)